.github/workflows/convert.yml

```diff
@@ -0,0 +1,56 @@
+name: convert precommit and test
+run-name: convert precommit and test
+on:
+  pull_request:
+    paths:
+      - convert/**
+      - .github/workflows/convert.yml
+jobs:
+  convert-pre-commit-actions:
```

Comment:
I would call this step something like `convert-lint`, `convert-pre-commit-actions` doesn't run pre-commit.
From my perspective we should leave only one job, where we:
1. Install dependencies (build)
2. Run pre-commit on all files (lint)
3. Run tests (test)

Why we should use pre-commit on the second stage? It simplifies a lot current workflow and it takes existing pre-commit configuration and its hooks (we won't have collisions of possible different configurations)
We also do not have `mypy` in our [pre-commit-config](https://github.com/epam/badgerdoc/blob/main/convert/.pre-commit-config.yaml) and workflow. Could you add it please?

Here's a good example of pre commit config you could take. And I think we should [remove commitizen](https://github.com/epam/badgerdoc/blob/main/convert/.pre-commit-config.yaml#L23) from current pre-commit config for convert since it's not not used in workflow and we have pending PR to add similar validation for the whole repos
```yaml
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
    -   id: trailing-whitespace
    -   id: end-of-file-fixer
    -   id: check-yaml
  - repo: https://github.com/pycqa/isort
    rev: 5.12.0
    hooks:
      - id: isort
        args:
          - --profile=black
          - --line-length=79
  - repo: https://github.com/psf/black
    rev: 23.1.0
    hooks:
      - id: black
        language_version: python3
        args:
          - --line-length=79
  - repo: https://github.com/pre-commit/mirrors-mypy
    rev: v1.1.1
    hooks:
      - id: mypy
        name: mypy
        entry: mypy
        language: python
        types: [python]
        require_serial: true
        args:
          - --strict
          - --ignore-missing-imports
          - --allow-untyped-decorators
          - --python-version=3.8
  - repo: https://github.com/pycqa/pylint
    rev: v2.17.1
    hooks:
        - id: pylint
          types: [ python ]
          args:
            - --max-line-length=79
            - --errors-only
            - --disable=E0401
```

Reply:
As I got, you mentioned this PR https://github.com/epam/badgerdoc/issues/194